### ISSUE TITLE
Potential incorrect ktlint wrapping violation on a when conditional without `{ }`

### ISSUE THREAD

**Issue by ZakTaccardi** [pinterest/ktlint#1839]
When running `ktlint` (`0.48.2`) over the following example) 

```kt
package zakTest

import zakTest.DayOfWeek.MONDAY
import zakTest.DayOfWeek.TUESDAY

internal class KotlinFormattingTest {

    fun dayOfWeek(dayOfWeek: DayOfWeek) {
        when (dayOfWeek) {
            MONDAY -> dayOfWeek.name
                .length
                .inc()
                .inc()
                .inc()
                .inc()
                .inc()
                .inc()
                .inc()
                .inc()
                .inc()
                .inc()
                .inc()

            TUESDAY -> dayOfWeek.name
                .length
                .inc()
                .inc()
                .inc()
                .inc()
                .inc()
                .inc()
                .inc()
                .inc()
                .inc()
                .inc()
                .inc()
                .inc()
                .inc()
        }
    }
}

internal enum class DayOfWeek {
    MONDAY,
    TUESDAY
}
```

It has the following `wrapping` violation.

```txt
KotlinFormattingTest.kt:10:21: Missing newline after "->" (wrapping)
KotlinFormattingTest.kt:24:22: Missing newline after "->" (wrapping)
```

formatting it changes the code to:

```kt
package zakTest

import zakTest.DayOfWeek.MONDAY
import zakTest.DayOfWeek.TUESDAY

internal class KotlinFormattingTest {

    fun dayOfWeek(dayOfWeek: DayOfWeek) {
        when (dayOfWeek) {
            MONDAY ->
                dayOfWeek.name
                    .length
                    .inc()
                    .inc()
                    .inc()
                    .inc()
                    .inc()
                    .inc()
                    .inc()
                    .inc()
                    .inc()
                    .inc()
                    .inc()

            TUESDAY ->
                dayOfWeek.name
                    .length
                    .inc()
                    .inc()
                    .inc()
                    .inc()
                    .inc()
                    .inc()
                    .inc()
                    .inc()
                    .inc()
                    .inc()
                    .inc()
                    .inc()
                    .inc()
        }
    }
}

internal enum class DayOfWeek {
    MONDAY,
    TUESDAY
}
```

This seems incorrect to me. My guess is that `ktlint` is calculating that `dayOfWeek.name.inc().inc().inc().inc().inc()` (etc) is breaking the line limit, so it wraps it to a new line. However, `TUESDAY -> dayOfWeek.name` is well under the line limit

**Comment by paul-dingemans** [comment]
> My guess is that ktlint is calculating that dayOfWeek.name.inc().inc().inc().inc().inc() (etc) is breaking the line limit, so it wraps it to a new line. However, TUESDAY -> dayOfWeek.name is well under the line limit

This guess is incorrect. The dot qualified expression `dayOfWeek.name.length...` is written as a multiline expression. Whenever an arrow in a when statement is followed by a multiline expression, that entire expression is wrapped to the next line.
